idxmgmt/ilm/supporter_factory.go

```diff
@@ -18,14 +18,67 @@
 package ilm
 
 import (
+	"time"
+
+	"github.com/pkg/errors"
+
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
-	"github.com/elastic/beats/libbeat/idxmgmt/ilm"
+	"github.com/elastic/beats/libbeat/common/fmtstr"
+	libilm "github.com/elastic/beats/libbeat/idxmgmt/ilm"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
 // MakeDefaultSupporter creates the ILM supporter for APM that is passed to libbeat.
-// Currently ILM is disabled by using a NoopSupport.
-func MakeDefaultSupporter(_ *logp.Logger, info beat.Info, config *common.Config) (ilm.Supporter, error) {
-	return ilm.NoopSupport(info, config)
+func MakeDefaultSupporter(log *logp.Logger, info beat.Info, cfg *common.Config) (libilm.Supporter, error) {
+	if log == nil {
+		log = logp.NewLogger("ilm")
+	} else {
+		log = log.Named("ilm")
+	}
+
+	var ilmCfg Config
+	if err := cfg.Unpack(&ilmCfg); err != nil {
+		return nil, err
+	}
+	aliasName, err := applyStaticFmtstr(info, ilmCfg.Alias)
+	if err != nil {
+		return nil, err
+	}
+	policyName, err := applyStaticFmtstr(info, ilmCfg.Policy)
+	if err != nil {
+		return nil, err
+	}
+
+	p, ok := eventPolicies[ilmCfg.Event]
+	if !ok {
+		return nil, errors.Errorf("policy for %s undefined", ilmCfg.Event)
+	}
+	policy := libilm.Policy{
+		Name: policyName,
+		Body: p,
+	}
+	alias := libilm.Alias{
+		Name:    aliasName,
+		Pattern: pattern,
+	}
+
+	return libilm.NewDefaultSupport(log, libilm.ModeEnabled, alias, policy, false, true), nil
+}
+
+func applyStaticFmtstr(info beat.Info, fmt *fmtstr.EventFormatString) (string, error) {
+	return fmt.Run(&beat.Event{
+		Fields: common.MapStr{
+			// beat object was left in for backward compatibility reason for older configs.
```

Comment:
would this be removed in the next major then?

Reply:
Yes, I'd definitly remove then. I was a bit torn whether I should add it first place or not as ILM is newly introduced, but thought I align with the rest of the libbeat code for this.